src/bacnet/basic/object/trendlog.c

```diff
@@ -1545,6 +1546,100 @@ int TL_encode_entry(uint8_t *apdu, int iLog, int iEntry)
     return (iLen);
 }
 
+
+int rr_decode_trendlog_entries(uint8_t *apdu, int apdu_len, BACNET_TRENDLOG_RECORD *rec) {
+    int len;
+    uint8_t tag_number = 0;
+    uint32_t len_value_type = 0;
+    bool first = true;
+    while (apdu_len > 0) {
+        if (IS_CONTEXT_SPECIFIC(apdu[0]) &&
+	      (decode_is_opening_tag_number(apdu, 0))) {
+	      if (!first) {
+	         rec->next = calloc(sizeof(BACNET_TRENDLOG_RECORD), 1);
```

Comment:
Another linked list pattern we use in this library is to use a fixed size array and pre-link the array entry next to point to the next array element.  Maybe this can support both on only calloc if the rec->next is NULL?